From 0a911e527a627f0dc709711fe7602a3ac2aad17e Mon Sep 17 00:00:00 2001
From: Jessica Yu <jeyu@kernel.org>
Date: Wed, 30 May 2018 10:46:28 +0200
Subject: [PATCH 5/5] modsign: print module name along with error message
Patch-mainline: No, submitted https://lkml.kernel.org/r/20180530090830.20737-4-jeyu@kernel.org
References: bsc#1093666

It is useful to know which module failed signature verification, so
print the module name along with the error message.

Note: SLE11 doesn't use kernel_is_locked_down(), but rather
secure_modules(), which checks sig_enforce, hence the difference from the
submitted patch upstream (and from the SLE15 version of this patch).

Signed-off-by: Jessica Yu <jeyu@kernel.org>
Acked-by: Jessica Yu <jeyu@suse.de>
---
 kernel/module.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/kernel/module.c b/kernel/module.c
index 1b05dc5d02c2..c9a7693cbccc 100644
--- a/kernel/module.c
+++ b/kernel/module.c
@@ -2451,7 +2451,8 @@ static int module_sig_check(struct load_info *info)
 		reason = "Loading of module with unavailable key";
 	decide:
 		if (secure_modules()) {
-			pr_notice("%s is rejected\n", reason);
+			pr_notice("%s: %s is rejected\n",
+				  info->mod->name, reason);
 			return -EKEYREJECTED;
 		}
 		return 0;
-- 
2.16.4

